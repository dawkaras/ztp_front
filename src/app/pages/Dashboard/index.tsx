import * as React from 'react';
import { Helmet } from 'react-helmet-async';

export function DashboardPage() {
  return (
    <>
      <Helmet>
        <title>Dashboard</title>
        <meta name="description" content="TEMPLATE" />
      </Helmet>
      <span>Dashboard container</span>
    </>
  );
}
