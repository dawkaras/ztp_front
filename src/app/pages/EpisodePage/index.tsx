import * as React from 'react';
import { useState, useEffect } from 'react';
import { Helmet } from 'react-helmet-async';
import { Container, Form, Col, Row, Tabs, Tab, Table } from 'react-bootstrap';
import ReactStars from 'react-rating-stars-component';
import { useHistory, useParams } from 'react-router-dom';
import Button from 'app/components/Button';
import uuid from 'react-uuid';

export function EpisodePage() {
  const history = useHistory();
  const { id } = useParams<{ id: string }>();
  const user = JSON.parse(localStorage.getItem('user') || '{}');
  const [loading, setLoading] = useState(true);
  const [seasons, setSeasons] = useState([
    { id: '0', seriesId: '0', number: '', seriesTitle: '' },
  ]);
  const [episode, setEpisode] = useState({
    id: '0',
    title: '',
    description: '',
    reviewsCount: 0,
    avgRate: 0,
    seasonId: '0',
    seriesId: '0',
    reviews: [],
    duration: 0,
  });
  useEffect(() => {
    if (user.role == null) history.push('/login');
    let requestOptions = {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json',
        Authorization: 'Bearer ' + user.token,
      },
    };
    fetch('https://localhost:5001/season', requestOptions)
      .then(response => {
        if (response.ok) {
          response.json().then(result => {
            setSeasons(result);
            if (id !== '0') {
              getEpisode(id);
            } else setLoading(false);
          });
        } else {
          response.json().then(result => alert(result.message));
        }
      })
      .catch(error => {
        alert(error);
      });
  }, [history, id, user.role, user.token]);
  const getEpisode = id => {
    let requestOptions = {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json',
        Authorization: 'Bearer ' + user.token,
      },
    };
    fetch('https://localhost:5001/episode/' + id, requestOptions)
      .then(response => {
        if (response.ok) {
          response.json().then(result => {
            setEpisode(result);
            setLoading(false);
          });
        } else {
          response.json().then(result => alert(result.message));
        }
      })
      .catch(error => {
        setLoading(false);
        alert(error);
      });
  };
  const renderCommentsTableData = () => {
    return episode.reviews.map((item, index) => {
      const { id, rate, review, userName } = item;
      return (
        <tr
          key={id}
          style={{ cursor: 'pointer' }}
          onClick={() =>
            history.push('/episodeReview/' + id, {
              title: episode.title,
              seriesTitle: seasons.find(x => x.id === episode.seasonId)
                ?.seriesTitle,
              seasonNumber: seasons.find(x => x.id === episode.seasonId)
                ?.number,
              episodeId: episode.id,
            })
          }
        >
          <td>{index + 1}</td>
          <td>{rate}</td>
          <td>{review}</td>
          <td>{userName}</td>
          <td
            style={{ visibility: user.role === 'admin' ? 'visible' : 'hidden' }}
            onClick={event => event.stopPropagation()}
          >
            <Button text="Delete" onClick={() => deleteReview(id)} />
          </td>
        </tr>
      );
    });
  };
  const save = event => {
    event.preventDefault();
    if (id === '0') episode.id = uuid();
    episode.title = event.target.elements.formTitle.value;
    episode.description = event.target.elements.formDescription.value;
    episode.seasonId = event.target.elements.formSeason.value;
    episode.duration = event.target.elements.formDuration.value;
    let requestOptions = {
      method: id === '0' ? 'POST' : 'PUT',
      headers: {
        'Content-Type': 'application/json',
        Authorization: 'Bearer ' + user.token,
      },
      body: JSON.stringify(episode),
    };
    fetch('https://localhost:5001/episode', requestOptions)
      .then(response => {
        if (response.ok) {
          if (id === '0') history.push('/episodes');
        } else response.json().then(result => alert(result.message));
      })
      .catch(error => {
        alert(error);
      });
  };
  const deleteReview = id => {
    let requestOptions = {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json',
        Authorization: 'Bearer ' + user.token,
      },
    };
    fetch('https://localhost:5001/episodeReview?id=' + id, requestOptions)
      .then(response => {
        if (response.ok) getEpisode(episode.id);
        else
          response
            .json()
            .then(result => alert(result.message))
            .catch(err =>
              alert(
                'Nie możesz usunąć komentarza dodanego przez innego użytkownika!'
              )
            );
      })
      .catch(error => {
        alert(error);
      });
  };
  return !loading ? (
    <>
      <Helmet>
        <title>Episode</title>
        <meta name="description" content="TEMPLATE" />
      </Helmet>
      <Container fluid>
        <div
          style={{
            background: 'white',
            width: '100%',
            margin: '3px',
            padding: '3px',
            textAlign: 'center',
            float: 'left',
            height: '800px',
            position: 'relative',
            marginTop: '50px',
          }}
        >
          <Form onSubmit={save}>
            <Form.Group as={Row} className="mb-3" controlId="formTitle">
              <Form.Label column sm="2">
                Title
              </Form.Label>
              <Col sm="8">
                <Form.Control
                  type="text"
                  defaultValue={episode.title}
                  readOnly={user.role === 'admin' ? false : true}
                />
              </Col>
            </Form.Group>
            <Form.Group as={Row} className="mb-3" controlId="formSeason">
              <Form.Label column sm="2">
                Series - Season
              </Form.Label>
              <Col sm="8">
                <Form.Select
                  defaultValue={episode.seasonId}
                  disabled={user.role !== 'admin' ? true : false}
                >
                  {seasons.map((item, index) => {
                    const { id, number, seriesTitle } = item;
                    return (
                      <option
                        key={id}
                        value={id}
                        label={seriesTitle + ' Season ' + number}
                      />
                    );
                  })}
                </Form.Select>
              </Col>
            </Form.Group>
            <Form.Group as={Row} className="mb-3" controlId="formDescription">
              <Form.Label column sm="2">
                Description
              </Form.Label>
              <Col sm="8">
                <Form.Control
                  as="textarea"
                  defaultValue={episode.description}
                  readOnly={user.role === 'admin' ? false : true}
                  rows={6}
                />
              </Col>
            </Form.Group>
            <Form.Group as={Row} className="mb-3" controlId="formDuration">
              <Form.Label column sm="2">
                Duration
              </Form.Label>
              <Col sm="8">
                <Form.Control
                  type="number"
                  defaultValue={episode.duration}
                  readOnly={user.role === 'admin' ? false : true}
                />
              </Col>
            </Form.Group>
            <div
              style={{
                visibility: user.role === 'admin' ? 'visible' : 'hidden',
              }}
            >
              <Button text="Save" />
            </div>
          </Form>
          <span
            style={{
              position: 'absolute',
              top: '5px',
              right: '10%',
              visibility: id !== '0' ? 'visible' : 'hidden',
            }}
          >
            <ReactStars
              count={1}
              size={40}
              activeColor="#ffd700"
              value={1}
              edit={false}
            />
          </span>
          <span
            style={{
              position: 'absolute',
              top: '18px',
              right: '8.5%',
              fontSize: 24,
              visibility: id !== '0' ? 'visible' : 'hidden',
            }}
          >
            {episode.avgRate.toFixed(1)}
          </span>
          <span
            style={{
              position: 'absolute',
              top: '50px',
              right: '6.5%',
              fontSize: 20,
              visibility: id !== '0' ? 'visible' : 'hidden',
            }}
          >
            Reviews count: {episode.reviewsCount}
          </span>
          <Tabs
            defaultActiveKey="profile"
            id="uncontrolled-tab-example"
            className="mb-3"
            style={{
              marginTop: '100px',
              visibility: id !== '0' ? 'visible' : 'hidden',
            }}
          >
            <Tab eventKey="reviews" title="Reviews">
              <Table style={{ visibility: id !== '0' ? 'visible' : 'hidden' }}>
                <tbody>
                  <tr>
                    <th>Id</th>
                    <th>Rate</th>
                    <th>Review</th>
                    <th>User</th>
                    <th
                      style={{
                        visibility:
                          user.role === 'admin' && id !== '0'
                            ? 'visible'
                            : 'hidden',
                      }}
                    >
                      Delete
                    </th>
                  </tr>
                  {renderCommentsTableData()}
                </tbody>
              </Table>
            </Tab>
          </Tabs>
          <div
            style={{
              position: 'absolute',
              top: '10%',
              right: '8%',
              visibility: id !== '0' ? 'visible' : 'hidden',
            }}
          >
            <Button
              text="Add review"
              onClick={() =>
                history.push('/episodeReview/0', {
                  title: episode.title,
                  seriesTitle: seasons.find(x => x.id === episode.seasonId)
                    ?.seriesTitle,
                  seasonNumber: seasons.find(x => x.id === episode.seasonId)
                    ?.number,
                  episodeId: episode.id,
                })
              }
            />
          </div>
        </div>
      </Container>
    </>
  ) : (
    <div></div>
  );
}
