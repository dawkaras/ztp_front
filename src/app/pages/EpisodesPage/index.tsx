import * as React from 'react';
import { useState, useEffect } from 'react';
import { Helmet } from 'react-helmet-async';
import { Container, Table } from 'react-bootstrap';
import { useHistory } from 'react-router-dom';
import Button from 'app/components/Button';

export function EpisodesPage() {
  const history = useHistory();
  const [episodes, setEpisodes] = useState<episode[]>([]);
  const user = JSON.parse(localStorage.getItem('user') || '{}');
  class episode {
    constructor(
      public id: number,
      public seriesTitle: string,
      public seasonNumber: string,
      public title: string,
      public avgRate: number,
      public reviewsCount: number
    ) {}
  }
  useEffect(() => {
    if (user.role == null) history.push('/login');
    let requestOptions = {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json',
        Authorization: 'Bearer ' + user.token,
      },
    };
    fetch('https://localhost:5001/episode', requestOptions)
      .then(response => {
        if (response.ok) {
          response.json().then(result => {
            setEpisodes(result);
          });
        } else {
          response.json().then(result => alert(result.message));
        }
      })
      .catch(error => {
        alert(error);
      });
  }, [history, user.role, user.token]);
  const deleteEpisode = id => {
    let requestOptions = {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json',
        Authorization: 'Bearer ' + user.token,
      },
    };
    fetch('https://localhost:5001/episode?id=' + id, requestOptions)
      .then(response => {
        if (response.ok) {
          let requestOptions = {
            method: 'GET',
            headers: {
              'Content-Type': 'application/json',
              Authorization: 'Bearer ' + user.token,
            },
          };
          fetch('https://localhost:5001/episode', requestOptions)
            .then(response => {
              if (response.ok) {
                response.json().then(result => {
                  setEpisodes(result);
                });
              } else {
                response.json().then(result => alert(result.message));
              }
            })
            .catch(error => {
              alert(error);
            });
        } else {
          response.json().then(result => alert(result.message));
        }
      })
      .catch(error => {
        alert(error);
      });
  };
  const renderTableData = () => {
    return episodes.map((item, index) => {
      const { id, seriesTitle, seasonNumber, title, avgRate, reviewsCount } =
        item;
      return (
        <tr
          key={id}
          style={{ cursor: 'pointer' }}
          onClick={() => history.push('/episode/' + id)}
        >
          <td>{index + 1}</td>
          <td>{seriesTitle}</td>
          <td>{seasonNumber}</td>
          <td>{title}</td>
          <td>{avgRate}</td>
          <td>{reviewsCount}</td>
          <td
            style={{ visibility: user.role === 'admin' ? 'visible' : 'hidden' }}
            onClick={event => event.stopPropagation()}
          >
            <Button text="Delete" onClick={() => deleteEpisode(id)} />
          </td>
        </tr>
      );
    });
  };
  return (
    <>
      <Helmet>
        <title>Episodes</title>
        <meta name="description" content="TEMPLATE" />
      </Helmet>
      <Container fluid>
        <div
          style={{
            height: '75px',
            background: 'white',
            width: '100%',
            top: '56px',
            position: 'fixed',
          }}
        >
          <div
            style={{
              position: 'absolute',
              right: '50px',
              top: '5px',
              visibility: user.role === 'admin' ? 'visible' : 'hidden',
            }}
          >
            <Button
              text="Add episode"
              onClick={() => history.push('/episode/0')}
            />
          </div>
        </div>
        <Table style={{ marginTop: '150px', marginBottom: '100px' }}>
          <tbody>
            <tr>
              <th>Id</th>
              <th>Series</th>
              <th>Season</th>
              <th>Title</th>
              <th>Rate</th>
              <th>Reviews</th>
              <th
                style={{
                  visibility: user.role === 'admin' ? 'visible' : 'hidden',
                }}
              >
                Delete
              </th>
            </tr>
            {renderTableData()}
          </tbody>
        </Table>
      </Container>
    </>
  );
}
