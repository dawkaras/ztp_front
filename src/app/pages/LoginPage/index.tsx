/**
 *
 * LoginPage
 *
 */

import React from 'react';

import LoginForm from 'app/components/LoginForm';

import { useHistory } from 'react-router-dom';
import { Container, Row, Col } from 'react-bootstrap';
import Button from 'app/components/Button';

export const LoginPage = () => {
  const history = useHistory();
  if (localStorage.getItem('user') != null) history.push('/movies');
  const goLogin = event => {
    event.preventDefault();
    const username = event.target.elements.formBasicEmail.value;
    const password = event.target.elements.formBasicPassword.value;
    let requestOptions = {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({ email: username, password }),
    };
    fetch('https://localhost:5001/login', requestOptions)
      .then(response => {
        if (response.ok) {
          response.json().then(result => {
            localStorage.setItem(
              'user',
              JSON.stringify({
                token: result.token,
                role: result.role,
                username,
              })
            );
            setTimeout(function () {
              history.push('/movies');
            }, 100);
          });
        } else {
          response.json().then(result => alert(result.message));
        }
      })
      .catch(error => {
        alert(error);
        history.push('/invalidCredentials');
      });
  };

  return (
    <Container fluid>
      <Row className="justify-content-md-center" style={{ marginTop: '55px' }}>
        <Col lg={6}>
          <LoginForm onSubmit={goLogin} text="Login!" />
          <span>You don't have an account? </span>
          <Button text="Register!" onClick={() => history.push('/register')} />
        </Col>
      </Row>
    </Container>
  );
};
