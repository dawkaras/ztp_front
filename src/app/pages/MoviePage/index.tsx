import * as React from 'react';
import { useState, useEffect } from 'react';
import { Helmet } from 'react-helmet-async';
import { Container, Form, Col, Row, Tabs, Tab, Table } from 'react-bootstrap';
import ReactStars from 'react-rating-stars-component';
import { useHistory, useParams } from 'react-router-dom';
import Button from 'app/components/Button';
import uuid from 'react-uuid';

export function MoviePage() {
  const history = useHistory();
  const [genres, setGenres] = useState([{ id: 0, name: '' }]);
  const [loading, setLoading] = useState(true);
  const [movie, setMovie] = useState({
    id: '0',
    title: '',
    description: '',
    duration: 0,
    director: '',
    reviewsCount: -1,
    avgRate: -1,
    reviews: [],
    genreId: '2',
  });
  const { id } = useParams<{ id: string }>();
  const user = JSON.parse(localStorage.getItem('user') || '{}');
  const getMovie = id => {
    let requestOptions = {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json',
        Authorization: 'Bearer ' + user.token,
      },
    };
    fetch('https://localhost:5001/movie/' + id, requestOptions)
      .then(response => {
        if (response.ok) {
          response.json().then(result => {
            setMovie(result);
            setLoading(false);
          });
        } else {
          response.json().then(result => alert(result.message));
        }
      })
      .catch(error => {
        setLoading(false);
        alert(error);
      });
  };
  useEffect(() => {
    if (user.role == null) history.push('/login');
    let requestOptions = {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json',
        Authorization: 'Bearer ' + user.token,
      },
    };
    fetch('https://localhost:5001/genre/', requestOptions)
      .then(response => {
        if (response.ok) {
          response.json().then(result => {
            setGenres(result);
          });
        } else {
          response.json().then(result => alert(result.message));
        }
      })
      .catch(error => {
        alert(error);
      });
    if (id !== '0') {
      getMovie(id);
    } else setLoading(false);
  }, [history, id, user.role, user.token]);
  const save = event => {
    event.preventDefault();
    if (id === '0') movie.id = uuid();
    movie.title = event.target.elements.formTitle.value;
    movie.director = event.target.elements.formDirector.value;
    movie.description = event.target.elements.formDescription.value;
    movie.duration = event.target.elements.formDuration.value;
    movie.genreId = event.target.elements.formGenre.value;
    let requestOptions = {
      method: id === '0' ? 'POST' : 'PUT',
      headers: {
        'Content-Type': 'application/json',
        Authorization: 'Bearer ' + user.token,
      },
      body: JSON.stringify(movie),
    };
    fetch('https://localhost:5001/movie', requestOptions)
      .then(response => {
        if (response.ok) {
          if (id === '0') history.push('/movies');
        } else {
          response.json().then(result => alert(result.message));
        }
      })
      .catch(error => {
        alert(error);
      });
  };
  const deleteReview = id => {
    let requestOptions = {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json',
        Authorization: 'Bearer ' + user.token,
      },
    };
    fetch('https://localhost:5001/movieReview?id=' + id, requestOptions)
      .then(response => {
        if (response.ok) {
          getMovie(movie.id);
        } else
          response
            .json()
            .then(result => alert(result.message))
            .catch(err =>
              alert(
                'Nie możesz usunąć komentarza dodanego przez innego użytkownika!'
              )
            );
      })
      .catch(error => {
        alert(error);
      });
  };
  const renderTableData = () => {
    return movie.reviews.map((item, index) => {
      const { id, rate, review, userName } = item;
      return (
        <tr
          key={id}
          style={{ cursor: 'pointer' }}
          onClick={() =>
            history.push('/movieReview/' + id, {
              title: movie.title,
              movieId: movie.id,
            })
          }
        >
          <td>{index + 1}</td>
          <td>{rate}</td>
          <td>{review}</td>
          <td>{userName}</td>
          <td
            style={{
              visibility:
                user.role === 'admin' && id !== '0' ? 'visible' : 'hidden',
            }}
            onClick={event => event.stopPropagation()}
          >
            <Button text="Delete" onClick={() => deleteReview(id)} />
          </td>
        </tr>
      );
    });
  };
  return !loading ? (
    <>
      <Helmet>
        <title>Movie</title>
        <meta name="description" content="TEMPLATE" />
      </Helmet>
      <Container fluid>
        <div
          style={{
            background: 'white',
            width: '100%',
            margin: '3px',
            padding: '3px',
            textAlign: 'center',
            float: 'left',
            height: '800px',
            position: 'relative',
            marginTop: '50px',
          }}
        >
          <Form onSubmit={save}>
            <Form.Group as={Row} className="mb-3" controlId="formTitle">
              <Form.Label column sm="2">
                Title
              </Form.Label>
              <Col sm="8">
                <Form.Control
                  type="text"
                  defaultValue={movie.title}
                  readOnly={user.role === 'admin' ? false : true}
                />
              </Col>
            </Form.Group>
            <Form.Group as={Row} className="mb-3" controlId="formDirector">
              <Form.Label column sm="2">
                Director
              </Form.Label>
              <Col sm="8">
                <Form.Control
                  type="text"
                  defaultValue={movie.director}
                  readOnly={user.role === 'admin' ? false : true}
                />
              </Col>
            </Form.Group>
            <Form.Group as={Row} className="mb-3" controlId="formGenre">
              <Form.Label column sm="2">
                Genre
              </Form.Label>
              <Col sm="8">
                <Form.Select
                  defaultValue={movie.genreId}
                  disabled={user.role !== 'admin' ? true : false}
                >
                  {genres.map((item, index) => {
                    const { id, name } = item;
                    return <option key={id} value={id} label={name} />;
                  })}
                </Form.Select>
              </Col>
            </Form.Group>
            <Form.Group as={Row} className="mb-3" controlId="formDescription">
              <Form.Label column sm="2">
                Description
              </Form.Label>
              <Col sm="8">
                <Form.Control
                  as="textarea"
                  defaultValue={movie.description}
                  readOnly={user.role === 'admin' ? false : true}
                  rows={6}
                />
              </Col>
            </Form.Group>
            <Form.Group as={Row} className="mb-3" controlId="formDuration">
              <Form.Label column sm="2">
                Duration
              </Form.Label>
              <Col sm="8">
                <Form.Control
                  type="number"
                  defaultValue={movie.duration}
                  readOnly={user.role === 'admin' ? false : true}
                />
              </Col>
            </Form.Group>
            <div
              style={{
                visibility: user.role === 'admin' ? 'visible' : 'hidden',
              }}
            >
              <Button text="Save" />
            </div>
          </Form>
          <span
            style={{
              position: 'absolute',
              top: '5px',
              right: '10%',
              visibility: id !== '0' ? 'visible' : 'hidden',
            }}
          >
            <ReactStars
              count={1}
              size={40}
              activeColor="#ffd700"
              value={1}
              edit={false}
            />
          </span>
          <span
            style={{
              position: 'absolute',
              top: '18px',
              right: '8.5%',
              fontSize: 24,
              visibility: id !== '0' ? 'visible' : 'hidden',
            }}
          >
            {movie.avgRate.toFixed(1)}
          </span>
          <span
            style={{
              position: 'absolute',
              top: '50px',
              right: '6.5%',
              fontSize: 20,
              visibility: id !== '0' ? 'visible' : 'hidden',
            }}
          >
            Reviews count: {movie.reviewsCount}
          </span>
          <Tabs
            defaultActiveKey="profile"
            id="uncontrolled-tab-example"
            className="mb-3"
            style={{
              marginTop: '50px',
              visibility: id !== '0' ? 'visible' : 'hidden',
            }}
          >
            <Tab eventKey="reviews" title="Reviews">
              <Table
                style={{
                  marginBottom: '50px',
                  visibility: id !== '0' ? 'visible' : 'hidden',
                }}
              >
                <tbody>
                  <tr>
                    <th>Id</th>
                    <th>Rate</th>
                    <th>Review</th>
                    <th>User</th>
                    <th
                      style={{
                        visibility:
                          user.role === 'admin' && id !== '0'
                            ? 'visible'
                            : 'hidden',
                      }}
                    >
                      Delete
                    </th>
                  </tr>
                  {renderTableData()}
                </tbody>
              </Table>
            </Tab>
          </Tabs>
          <div
            style={{
              position: 'absolute',
              top: '10%',
              right: '8%',
              visibility:
                localStorage.getItem('user') !== null && id !== '0'
                  ? 'visible'
                  : 'hidden',
            }}
          >
            <Button
              text="Add review"
              onClick={() =>
                history.push('/movieReview/0', {
                  title: movie.title,
                  movieId: movie.id,
                })
              }
            />
          </div>
        </div>
      </Container>
    </>
  ) : (
    <div></div>
  );
}
