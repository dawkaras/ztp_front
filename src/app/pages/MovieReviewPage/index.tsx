/**
 *
 * ReviewPage
 *
 */

import React from 'react';
import { useEffect, useState } from 'react';
import { Container, Row, Col, Form } from 'react-bootstrap';
import ReactStars from 'react-rating-stars-component';
import Button from 'app/components/Button';
import { useHistory, useParams } from 'react-router-dom';
import uuid from 'react-uuid';

export const MovieReviewPage = props => {
  const history = useHistory();
  const user = JSON.parse(localStorage.getItem('user') || '{}');
  const { id } = useParams<{ id: string }>();
  const [rate, setRate] = useState(0);
  const [loading, setLoading] = useState(true);
  const [review, setReview] = useState({
    id: '0',
    rate: 0,
    review: '',
    userId: '',
    movieId: '',
    userName: '',
  });
  useEffect(() => {
    if (user.role == null) history.push('/login');
    if (id !== '0') {
      let requestOptions = {
        method: 'GET',
        headers: {
          'Content-Type': 'application/json',
          Authorization: 'Bearer ' + user.token,
        },
      };
      fetch('https://localhost:5001/movieReview/' + id, requestOptions)
        .then(response => {
          if (response.ok) {
            response.json().then(result => {
              setReview(result);
              setRate(result.rate);
              setLoading(false);
            });
          } else {
            response.json().then(result => alert(result.message));
          }
        })
        .catch(error => {
          alert(error);
          setLoading(false);
        });
    } else setLoading(false);
  }, [history, id, user.role, user.token]);
  const save = event => {
    event.preventDefault();
    if (id === '0') review.id = uuid();
    review.rate = rate;
    review.review = event.target.elements.formReview.value;
    review.movieId = props.location.state.movieId;
    let requestOptions = {
      method: id === '0' ? 'POST' : 'PUT',
      headers: {
        'Content-Type': 'application/json',
        Authorization: 'Bearer ' + user.token,
      },
      body: JSON.stringify(review),
    };
    fetch('https://localhost:5001/movieReview', requestOptions)
      .then(response => {
        if (response.ok) {
          if (id === '0') history.push('/movie/' + review.movieId);
        } else
          response
            .json()
            .then(result => alert(result.message))
            .catch(err =>
              alert(
                'Nie możesz edytować komentarza dodanego przez innego użytkownika!'
              )
            );
      })
      .catch(error => {
        alert(error);
      });
  };
  return !loading ? (
    <Container fluid>
      <Row className="justify-content-md-center" style={{ marginTop: '60px' }}>
        <Col lg={6}>
          <Form.Label>Review for {props.location.state.title}.</Form.Label>
        </Col>
      </Row>
      <Row className="justify-content-md-center">
        <Col lg={6}>
          <Form style={{ marginTop: '10px' }} onSubmit={save}>
            <Form.Group className="mb-3" controlId="formRate">
              <Form.Label>Rate:</Form.Label>

              <ReactStars
                count={5}
                size={32}
                activeColor="#ffd700"
                edit={localStorage.getItem('user') == null ? false : true}
                value={rate}
                onChange={value => setRate(value)}
              />
            </Form.Group>
            <Form.Group className="mb-3" controlId="formReview">
              <Form.Label>Review:</Form.Label>
              <Form.Control
                as="textarea"
                rows={6}
                disabled={localStorage.getItem('user') == null ? true : false}
                defaultValue={review.review}
              />
            </Form.Group>
            <Form.Group
              className="mb-3"
              controlId="formUser"
              style={{ visibility: id !== '0' ? 'visible' : 'hidden' }}
            >
              <Form.Label>Added by: {review.userName}</Form.Label>
            </Form.Group>
            <div
              style={{
                visibility:
                  localStorage.getItem('user') == null ? 'hidden' : 'visible',
              }}
            >
              <Button text="Add" />
            </div>
          </Form>
        </Col>
      </Row>
    </Container>
  ) : (
    <div></div>
  );
};
