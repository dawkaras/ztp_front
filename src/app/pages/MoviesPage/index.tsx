import * as React from 'react';
import { useState, useEffect } from 'react';
import { Helmet } from 'react-helmet-async';
import { Container, Table } from 'react-bootstrap';
import { useHistory } from 'react-router-dom';
import Button from 'app/components/Button';

export function MoviesPage() {
  const history = useHistory();
  const [movies, setMovies] = useState<movie[]>([]);
  const user = JSON.parse(localStorage.getItem('user') || '{}');
  class movie {
    constructor(
      public id: number,
      public title: string,
      public avgRate: number,
      public reviewsCount: number
    ) {}
  }
  const getMovies = () => {
    let requestOptions = {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json',
        Authorization: 'Bearer ' + user.token,
      },
    };
    fetch('https://localhost:5001/movie', requestOptions)
      .then(response => {
        if (response.ok) {
          response.json().then(result => {
            setMovies(result);
          });
        } else {
          response.json().then(result => alert(result.message));
        }
      })
      .catch(error => {
        alert(error);
      });
  };
  useEffect(() => {
    getMovies();
    if (user.role == null) history.push('/login');
  }, [history, user.role, user.token]);
  const deleteMovie = id => {
    let requestOptions = {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json',
        Authorization: 'Bearer ' + user.token,
      },
    };
    fetch('https://localhost:5001/movie?id=' + id, requestOptions)
      .then(response => {
        if (response.ok) getMovies();
        else
          response
            .json()
            .then(result =>
              alert(
                result.message +
                  '\n' +
                  'Pamiętaj, aby najpierw usunąć wszystkie recenzje filmu!'
              )
            );
      })
      .catch(error => {
        alert(error);
      });
  };
  const renderTableData = () => {
    return movies.map((movie, index) => {
      const { id, title, avgRate, reviewsCount } = movie;
      return (
        <tr
          key={id}
          style={{ cursor: 'pointer' }}
          onClick={() => history.push('/movie/' + id)}
        >
          <td>{index + 1}</td>
          <td>{title}</td>
          <td>{avgRate}</td>
          <td>{reviewsCount}</td>
          <td
            style={{ visibility: user.role === 'admin' ? 'visible' : 'hidden' }}
            onClick={event => event.stopPropagation()}
          >
            <Button
              text="Delete"
              onClick={() => {
                deleteMovie(id);
              }}
            />
          </td>
        </tr>
      );
    });
  };
  return (
    <>
      <Helmet>
        <title>Movies</title>
        <meta name="description" content="TEMPLATE" />
      </Helmet>
      <Container fluid>
        <div
          style={{
            height: '75px',
            background: 'white',
            width: '100%',
            top: '56px',
            position: 'fixed',
          }}
        >
          <div
            style={{
              position: 'absolute',
              right: '50px',
              top: '5px',
              visibility: user.role === 'admin' ? 'visible' : 'hidden',
            }}
          >
            <Button text="Add movie" onClick={() => history.push('/movie/0')} />
          </div>
        </div>
        <Table style={{ marginTop: '150px', marginBottom: '100px' }}>
          <tbody>
            <tr>
              <th>Id</th>
              <th>Title</th>
              <th>Rate</th>
              <th>Reviews</th>
              <th
                style={{
                  visibility: user.role === 'admin' ? 'visible' : 'hidden',
                }}
              >
                Delete
              </th>
            </tr>
            {renderTableData()}
          </tbody>
        </Table>
      </Container>
    </>
  );
}
