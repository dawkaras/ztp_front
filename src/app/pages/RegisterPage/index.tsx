/**
 *
 * LoginPage
 *
 */

import React from 'react';

import { useHistory } from 'react-router-dom';
import { Container, Row, Col, Form } from 'react-bootstrap';
import Button from 'app/components/Button';

export const RegisterPage = () => {
  const history = useHistory();
  if (localStorage.getItem('user') != null) history.push('/movies');
  const register = event => {
    event.preventDefault();
    const username = event.target.elements.formBasicEmail.value;
    const password = event.target.elements.formBasicPassword.value;
    const confirmPassword =
      event.target.elements.formBasicConfirmPassword.value;
    if (confirmPassword == password) {
      let requestOptions = {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({ email: username, password }),
      };
      fetch('https://localhost:5001/signup', requestOptions)
        .then(response => {
          if (response.ok) {
            history.push('/login');
          } else {
            response.json().then(result => {
              alert(result.message);
            });
          }
        })
        .catch(error => {
          alert(error);
        });
    } else alert('Passwords do not match!');
  };

  return (
    <Container fluid>
      <Row className="justify-content-md-center">
        <Col lg={6}>
          <Form onSubmit={register} style={{ marginTop: '55px' }}>
            <Form.Group controlId="formBasicEmail">
              <Form.Label>Email:</Form.Label>
              <Form.Control type="text" placeholder="Type your login" />
            </Form.Group>
            <Form.Group controlId="formBasicPassword">
              <Form.Label>Password:</Form.Label>
              <Form.Control type="password" placeholder="Type your password:" />
            </Form.Group>
            <span>
              Password need at least 5 chars, 1 big letter, 1 number, 1 special
              sign
            </span>
            <Form.Group controlId="formBasicConfirmPassword">
              <Form.Label>Confirm password:</Form.Label>
              <Form.Control
                type="password"
                placeholder="Confirm your password:"
              />
            </Form.Group>
            <Button text="Register!" />
          </Form>
        </Col>
      </Row>
    </Container>
  );
};
