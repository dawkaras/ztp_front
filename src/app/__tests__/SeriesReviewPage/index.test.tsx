import * as React from 'react';
import renderer from 'react-test-renderer';
import { SeriesReviewPage } from 'app/pages/SeriesReviewPage';
import { MemoryRouter, Route } from 'react-router-dom';

it('renders SeriesReviewPage page', () => {
  const component = renderer
    .create(
      <MemoryRouter initialEntries={['/seriesReview/0']}>
        <Route path="/seriesReview/:id">
          <SeriesReviewPage
            props={{ title: 'Test1Title', seriesId: 'Test1SeriesId' }}
          />
        </Route>
      </MemoryRouter>
    )
    .toJSON();
  expect(component).toMatchSnapshot();
});
