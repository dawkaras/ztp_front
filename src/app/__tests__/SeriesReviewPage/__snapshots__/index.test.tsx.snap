// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SeriesReviewPage page 1`] = `<div />`;
